### ISSUE TITLE
[New feature] Ability to append labels instead of replace them all

### ISSUE THREAD

**Issue by BarthV** [prometheus/jmx_exporter#104]
Currently when we're using a rule to match some beans attributes, there's no simple way to append custom labels without romving existing ones.

Can you add this on the roadmap please :angel:  ?

Thanks a lot

**Comment by brian-brazil** [comment]
Can you give an example of what you're trying to do?

**Comment by automaticgiant** [comment]
i think i know what they are talking about. i want something like that.
the default pattern seems to extract any/all/arbitrary labels from `domain<beanpropertyName1=beanPropertyValue1, beanpropertyName2=beanPropertyValue2, ...><key1, key2, ...>attrName: value` however, if you would like to set up a consistant naming mapping from bean objectName to metric name, you have to cover all desired labels in your pattern. i've been unable to come up with a way to write a rule that can do so while preserving the default rule's ability to get all the labels.

**Comment by brian-brazil** [comment]
The default is doing some things that can't be done with just one rule (and also handles some escaping issues),  several rules are required if you're doing things more specifically.

**Comment by automaticgiant** [comment]
i was also (due to quitting early probably) unable to find where in the source the default behavior was defined. a pointer would be appreciated.
i'd really like to be able to reproduce default behavior in a config and tweak it. that might satisfy @BarthV as well.

**Comment by brian-brazil** [comment]
> so are you suggesting universally emulating default behavior is possible with a multiple rules?

Mostly. In practical terms you should almost always be able to produce a good config for a given server.

**Comment by automaticgiant** [comment]
i'm interested in a "universal"-ish config, instead of having to set up a config for each jmx app.
we are kafkaing right now, presumably zookeeper, mirrormaker, nifi, etc to follow...
my main complaint is finding a solution to the metric name and tag setup for instances where you have a bunch of attributes and it explodes the metric namespace with them.
otherwise, default rule would work fine.

have:
```
#kafka_server_BrokerTopicMetrics_OneMinuteRate{name="BytesOutPerSec",topic="SYSLOG",} 3.0799494822146203
#kafka_server_BrokerTopicMetrics_OneMinuteRate{name="FailedFetchRequestsPerSec",topic="SYSLOG",} 0.0
#kafka_server_BrokerTopicMetrics_OneMinuteRate{name="BytesInPerSec",topic="SYSLOG",} 1.4507932500151888
#kafka_server_BrokerTopicMetrics_OneMinuteRate{name="BytesRejectedPerSec",topic="SYSLOG",} 0.0
#kafka_server_BrokerTopicMetrics_OneMinuteRate{name="TotalFetchRequestsPerSec",topic="SYSLOG",} 63.990540940470744
#kafka_server_BrokerTopicMetrics_OneMinuteRate{name="FailedProduceRequestsPerSec",topic="SYSLOG",} 0.0
#kafka_server_BrokerTopicMetrics_OneMinuteRate{name="MessagesInPerSec",topic="SYSLOG",} 0.010284049118568367
#kafka_server_BrokerTopicMetrics_OneMinuteRate{name="TotalProduceRequestsPerSec",topic="SYSLOG",} 0.010284049118568367
#kafka_server_BrokerTopicMetrics_FiveMinuteRate{name="BytesOutPerSec",topic="SYSLOG",} 7.914210031726753
#kafka_server_BrokerTopicMetrics_FiveMinuteRate{name="FailedFetchRequestsPerSec",topic="SYSLOG",} 0.0
#kafka_server_BrokerTopicMetrics_FiveMinuteRate{name="BytesInPerSec",topic="SYSLOG",} 2.8756598718384767
#kafka_server_BrokerTopicMetrics_FiveMinuteRate{name="BytesRejectedPerSec",topic="SYSLOG",} 0.0
#kafka_server_BrokerTopicMetrics_FiveMinuteRate{name="TotalFetchRequestsPerSec",topic="SYSLOG",} 75.86600439041331
#kafka_server_BrokerTopicMetrics_FiveMinuteRate{name="FailedProduceRequestsPerSec",topic="SYSLOG",} 0.0
#kafka_server_BrokerTopicMetrics_FiveMinuteRate{name="MessagesInPerSec",topic="SYSLOG",} 0.016103353485893434
#kafka_server_BrokerTopicMetrics_FiveMinuteRate{name="TotalProduceRequestsPerSec",topic="SYSLOG",} 0.016103353485893434
#kafka_server_FetcherLagMetrics_Value{name="ConsumerLag",clientId="ReplicaFetcherThread-0-4",topic="SYSLOG",partition="4",} 0.0
#kafka_server_FetcherLagMetrics_Value{name="ConsumerLag",clientId="ReplicaFetcherThread-0-2",topic="SYSLOG",partition="7",} 0.0
#kafka_server_FetcherLagMetrics_Value{name="ConsumerLag",clientId="ReplicaFetcherThread-0-4",topic="SYSLOG",partition="9",} 0.0
#kafka_server_FetcherLagMetrics_Value{name="ConsumerLag",clientId="ReplicaFetcherThread-0-5",topic="SYSLOG",partition="0",} 0.0
#kafka_server_FetcherLagMetrics_Value{name="ConsumerLag",clientId="ReplicaFetcherThread-0-4",topic="SYSLOG",partition="3",} 0.0
#kafka_server_FetcherLagMetrics_Value{name="ConsumerLag",clientId="ReplicaFetcherThread-0-5",topic="SYSLOG",partition="8",} 0.0
#kafka_server_BrokerTopicMetrics_FifteenMinuteRate{name="BytesOutPerSec",topic="SYSLOG",} 9.285483891560915
#kafka_server_BrokerTopicMetrics_FifteenMinuteRate{name="FailedFetchRequestsPerSec",topic="SYSLOG",} 0.0
#kafka_server_BrokerTopicMetrics_FifteenMinuteRate{name="BytesInPerSec",topic="SYSLOG",} 3.078354766954514
#kafka_server_BrokerTopicMetrics_FifteenMinuteRate{name="BytesRejectedPerSec",topic="SYSLOG",} 0.0
#kafka_server_BrokerTopicMetrics_FifteenMinuteRate{name="TotalFetchRequestsPerSec",topic="SYSLOG",} 83.79707100258736
#kafka_server_BrokerTopicMetrics_FifteenMinuteRate{name="FailedProduceRequestsPerSec",topic="SYSLOG",} 0.0
#kafka_server_BrokerTopicMetrics_FifteenMinuteRate{name="MessagesInPerSec",topic="SYSLOG",} 0.015124558666446455
#kafka_server_BrokerTopicMetrics_FifteenMinuteRate{name="TotalProduceRequestsPerSec",topic="SYSLOG",} 0.015124558666446455
```
want something more like:
```
#kafka_server:BrokerTopicMetrics{attribute="Count...
#kafka_server:BrokerTopicMetrics{attribute="MeanRate...
#kafka_server:BrokerTopicMetrics{attribute="OneMinuteRate...
#kafka_server:BrokerTopicMetrics{attribute="FiveMinuteRate...
#kafka_server:BrokerTopicMetrics{attribute="FifteenMinuteRate...
```

basically, i just want to move the attribute into a tag, not as part of the metric name.

**Comment by automaticgiant** [comment]
sorry if i'm hijacking the issue or if it's a different issue. depending on solution, it could be the same issue. @BarthV

**Comment by brian-brazil** [comment]
> i'm interested in a "universal"-ish config, instead of having to set up a config for each jmx app.

That's unlikely to happen, JMX usage is nowhere near standard enough to allow for that. That's why it's all regexes, there's no common structure.

> basically, i just want to move the attribute into a tag, not as part of the metric name.

Those belong as part of the metric name, though of those you really only want the Count and Value.
